0.2.1

JeGoi opened the pull request:
Go to version 0.2.1
- Fix hostapd and wpa supplicant versions
- Hostapd: Remove kernel lines from hostapd install
- Hostapd: Add vlan file for dynamic vlan

JeGoi committed:
Remove kernel tasks

JeGoi committed:
add dynamic vlan file for hostapd

JeGoi committed:
Add vlan file for dynamic vlan

JeGoi committed:
Update version and changelog

JeGoi committed:
Fix a static hostapd version

JeGoi committed:
Change version variable name

JeGoi committed:
Add wpasupplicant version for debian

nqb commented on roles/hostapd/defaults/main.yml:
```diff
@@ -45,4 +46,5 @@ hostapd__configs:
     hostapd__wpa_key_mgmt: ''
     hostapd__wpa_pairwise: ''
-
+    #hostapd__vlan_file_content: ''
+    #hostapd__vlan_file: ''
```
I'm not sure Ansible will work if we keep defaults like this even if there is a conditional in `Add vlan_file for hostapd` task.

JeGoi replied:
I had them here to show all options. I could provide them with values that will not change the default config.

The idea is that if hostapd__vlan_file is there is needs to be defined and not empty. So, if the variable hostapd__vlan_file is defined but empty, it should failed. I will had a test of not empty.

The hostapd__vlan_file_content of the hostapd__vlan_file could be empty. It has a default mapping.
```
# If no entries are provided by this file, the station is statically mapped
# to <bss-iface>.<vlan-id> interfaces.
# Each line can optionally also contain the name of a bridge to add the VLAN to
 ```
[source: https://w1.fi/cgit/hostap/plain/hostapd/hostapd.conf ]

We also could add a test around dynamic_vlan. The dynamic_vlan is defined and > 0, then hostapd__vlan_file should be there and not empty.

nqb replied:
Looks good to me. I had a minor comment.

nqb commented on CHANGELOG.md:
```diff
@@ -5,6 +5,11 @@ All notable changes to this project will be documented in this file.
 
+## [v0.2.1]
+### Added
+- Remove kernel lines from hostapd install
```
`kmod` is not necessary anymore ? Could you add more details here ?

JeGoi replied:
Sure, it is not useful to have it there since the kernel module is in another role (mac80211hwsim) and is installed before hostapd. Kmod is not used by hostapd.

So, we should have move that the install from hostapd to part to mac80211hwsim.
Kmod is used to mount kernel modules but we used modprobe from https://docs.ansible.com/ansible/2.10/collections/community/general/modprobe_module.html to add modules. It also looks that modprobe is there by default on debian and centos... That is why I did not move installation of kmod from hostapd to mac80211hwsim role.

Does it answer your question?

JeGoi committed:
Move default config from deploy to main to avoid same task multiple time

JeGoi committed:
Add default values and link for default values
Add a debug to show vlan config if used
Change string to values for default variables

JeGoi committed:
Add default variables, now the hostapd all_available_values should work

JeGoi committed:
fix the vlan file for default varaibles

nqb commented on roles/hostapd/tasks/deploy_configs.yml:
```diff
@@ -6,8 +6,21 @@
-    src: hostapd_default.conf
-    dest: '/etc/hostapd/hostapd.conf'
+- name: Show vlan hostapd configuration if vlan used
+  debug:
```
```suggestion
  debug:
    verbosity: 1
```

nqb replied:
According to [debug module documentation](https://docs.ansible.com/ansible/latest/collections/ansible/builtin/debug_module.html), it will appear only if you run `ansible-playbook -v` (one -v)

JeGoi replied:
The main idea was to give some visible info for a quick review of the configuration in the context of using hostapd dynamic vlan.
If it comes to use ansible verbosity, then I am not sure it is still useful, isn't it?

nqb replied:
As you prefer. I see two options:
- we keep as is: display information if necessary
- using verbosity: display information only if you specify `-v` at ansible execution

Second option is a best practice I've seen in some Ansible official document

JeGoi replied:
Ok than lets do it that way.
Thanks

JeGoi committed:
Update roles/hostapd/tasks/deploy_configs.yml

Add verbostiy

Co-authored-by: Nicolas Quiniou-Briand <nqb+git@azyx.fr>
